src/endpoints/data_objects/impl.cpp

```diff
@@ -465,12 +505,45 @@ namespace
                         return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
                     }
 
+                    auto openmode = std::ios_base::out;
+
+                    if (const auto iter = _args.find("truncate"); iter != std::end(_args) && iter->second == "0") {
+                        openmode |= std::ios_base::in;
+                    }
+
+                    if (const auto iter = _args.find("append"); iter != std::end(_args) && iter->second == "1") {
+                        openmode |= std::ios_base::app;
+                    }
+
                     log::trace("{}: Opening data object [{}] for write.", fn, lpath_iter->second);
                     log::trace("{}: (write) Initializing for single buffer write.", fn);
 
                     conn = irods::get_connection(client_info->username);
+
+                    // Enable ticket if the request includes one.
+                    if (const auto iter = _args.find("ticket"); iter != std::end(_args)) {
+                        if (const auto ec = irods::enable_ticket(conn, iter->second); ec < 0) {
+                            res.result(http::status::internal_server_error);
+                            res.body() = json{
+                                {"irods_response", {
+                                    {"error_code", ec},
+                                    {"error_message", "Error enabling ticket on connection."}
+                                }}
+                            }.dump();
+                            res.prepare_payload();
+                            return _sess_ptr->send(std::move(res));
+                        }
+                    }
+
                     tp = std::make_unique<io::client::native_transport>(conn);
-                    out = std::make_unique<io::odstream>(*tp, lpath_iter->second);
+
+                    if (const auto iter = _args.find("resource"); iter != std::end(_args)) {
```

Comment:
do we need to consider hierarchies here?     probably already handled... but just leaving a note.

Reply:
So far, none of the operations support hierarchies. That's not to say they won't, no thought has been spent on supporting them yet.

Willing to discuss whether they should be supported though.

Reply:
no interest in supporting them.   just want to make sure this code works as desired when handed one.   aka, does it error out correctly/as expected?

Reply:
I'd expect either an error or undefined behavior in the context of iRODS.

Also, dstream doesn't support hierarchies.

Reply:
fair enough - resolving.